src/nested_pandas/datasets/generation.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+
+from nested_pandas import NestedFrame
+
+
+def generate_data(n_base, n_layer, seed=None) -> NestedFrame:
+    """Generates a toy dataset.
+
+    Parameters
+    ----------
+    n_base : int
+        The number of rows to generate for the base layer
+    n_layer : int, or dict
+        The number of rows per n_base row to generate for a nested layer.
+        Alternatively, a dictionary of layer label, layer_size pairs may be
+        specified to created multiple nested columns with custom sizing.
+    seed : int
+        A seed to use for random generation of data
+
+    Returns
+    -------
+    NestedFrame
+        The constructed NestedFrame.
+
+    Examples
+    --------
+    >>> nested_pandas.datasets.generate_data(10,100)
+    >>> nested_pandas.datasets.generate_data(10, {"nested_a": 100, "nested_b": 200})
+    """
+    # use provided seed, "None" acts as if no seed is provided
+    randomstate = np.random.RandomState(seed=seed)
+
+    # Generate base data
+    base_data = {"a": randomstate.random(n_base), "b": randomstate.random(n_base) * 2}
+    base_nf = NestedFrame(data=base_data)
+
+    # In case of int, create a single nested layer called "nested"
+    if isinstance(n_layer, int):
+        n_layer = {"nested": n_layer}
+
+    # It should be a dictionary
+    if isinstance(n_layer, dict):
+        for key in n_layer:
+            layer_size = n_layer[key]
+            layer_data = {
+                "t": randomstate.random(layer_size * n_base) * 20,
```

Comment:
Sorry, one more thing, let's sort by time here?
```suggestion
                "t": np.sort(randomstate.random(layer_size * n_base) * 20),
```